server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt

```diff
@@ -372,6 +381,39 @@ class HumanSkeleton(
 		viveEmulation.update()
 	}
 
+	/**
+	 * Enforce rotation constraints on all bones
+	 */
+	private fun enforceConstraints() {
+		if (!enforceConstraints) return
+
+		for (bone in allHumanBones) {
+			// Correct the rotation if it violates a constraint
+			val initialRot = bone.getGlobalRotation()
+			val newRot = bone.rotationConstraint.applyConstraint(initialRot, bone)
+			bone.setRotationRaw(newRot)
+			bone.updateThisNode()
+
+			if (!correctConstraints || bone.rotationConstraint.constraintType == ConstraintType.HINGE || bone.rotationConstraint.constraintType == ConstraintType.LOOSE_HINGE) continue
+
+			// Apply a correction to the tracker rotation if filtering is not greatly affecting the output rotation
+			val deltaRot = newRot * initialRot.inv()
+			val angle = deltaRot.angleR()
+			val tracker = getTrackerForBone(bone.boneType)
+			val parentTracker = getTrackerForBone(bone.parent?.boneType)
+			if ((angle > 0.01f) && (
+					tracker?.filteringHandler?.getFilteringImpact()
+						?: 1f
+					) < 0.01f && (
+					parentTracker?.filteringHandler?.getFilteringImpact()
+						?: 1f
+					) < 0.01f
```

Comment:
It may be a bit restrictive but I found in testing that values generally stay far below this when not moving around very fast. I will test a bit more and get back to you

Reply:
I increased the value to 2 degrees (compared to 0.6 degrees previously), which should allow correction more often without sacrificing much accuracy.